### ISSUE TITLE
Several remaining Bugs in Split Track Analysis (Release relevant)

### ISSUE THREAD

**Issue by njohnston** [osmandapp/Osmand#4041]
There are several problems in the updated track split analysis code. [Sample track available](https://drive.google.com/file/d/0ByKX_Fubbb4QUFNSQW9MZEtWMVU/view?usp=sharing).

### Split interval markers don't appear in "Analyse on Map" unless "Show on map" is set

If a split interval is set, markers are not shown in "Analyse on Map" unless "Show on map" is selected.

Split markers should always be shown in "Analyse on Map" if a split interval is selected. (Reasoning: I don't need to select "Show on map" to see the track itself on "Analyse on Map", so I shouldn't have to select "Show on map" to see split markers in "Analyse on Map".)

### There are duplicate labels

With a 1 km interval:

![split_duplicate_labels](https://user-images.githubusercontent.com/638458/27875211-53070eca-61aa-11e7-8c03-bafd67b93c7b.png)

I would expect each number to be shown only once.

### When splitting by time, labels can overlap at the start and end

In this case, the track starts and finishes at the same place. If I split by 15 minutes, two labels at the start/destination are displayed on top of each other:

![overlapping_labels](https://user-images.githubusercontent.com/638458/27875246-6a7341aa-61aa-11e7-9c98-217f8534ed40.png)

I don't understand the 26:54 label either. I would expect this to be either the start time (0) or the end time, which is much more than 26:54.

### What I'd like to see

Ideally I'd like to see splits shown like this:

![cardiotrainer_split_1_small](https://user-images.githubusercontent.com/638458/27875273-85701334-61aa-11e7-8f12-d6c0ab347cf9.png)

Having distance and time here is very useful. The label placement here is better because they don't obscure the track.

I'd also like to be able to see the times for each km or unit like this:

![cardiotrainer_split_2_small](https://user-images.githubusercontent.com/638458/27875295-9814273c-61aa-11e7-970d-599e34835a77.png)

This is very useful, particularly for long tracks.

**Comment by sonora** [comment]
Also:
- In our previous implementation of this the labeled split points were tappable and had a context menu
- Visibility: We now use the track color as background color also for these labels, but we always seem to use white as the font color. For bright track colors (in Touring view map style the default is cyan) this is very hard to read. we should auto-switch from white to black from a certain brightness threshold on.

**Comment by sonora** [comment]
One more issue I observe: Up until at least the nightly builds of 2017-07-10, when on the track analysis screen I tap the bottom-left "Split interval" button, nothing happens and I simply remain on the track analysis screen (other than the top-right "Share" button disappears). Only after I change the device orientation from portrait to landscape, does the new split interval screen come up (and remains visible even if successively going back to the portrait orientation).

**Comment by njohnston** [comment]
I have looked into the track split support again (app compiled today) and I've realised that split intervals are broken for tracks with multiple segments.

In such tracks, the splits are calculated for each segment **and** for the overall track. Imagine a track where I walk 6 km to a particular point, then start a new segment, and then walk back to the start following my original route. Three distinct sets of split information will be shown on the track:

* Splits for the outbound segment
* Splits for the return segment
* Splits for the whole trip (both segments combined into a single segment)

This means, for example, that 2 km before the turnaround point, there are three labels:

* 4 km (outbound)
* 2 km (return)
* 8 km (combined)

This is very confusing and not very obvious. Distance intervals should only be calculated on the combined track, **not** on each segment. Time intervals should be calculated on the combined track too, but I don't know how to account for pauses. (For example, if there's a pause of 20 minutes between two segments, and the track is being split at 10 minute intervals, how should this be handled?)

There is another problem with the track analysis screen. Each segment has a "Split interval" button:

![split_interval_buttons](https://user-images.githubusercontent.com/638458/28241366-1d65b508-698b-11e7-8bf4-cdb7921e47aa.jpg)

Each "Split interval" button always opens the same view (the entire track). This is different to the "Analyse on map" button which shows a map with the whole track but charts only for that segment.

I think it would be better if the "Split interval" button showed information for that segment only. Since a total segment is now automatically calculated, this allows the whole track to be analysed as a single track, and also allows each segment to be analysed, which is useful if you combine different modes of transport in the same track. (E.g. train journey in one segment, hiking in the next, cycling in the next, ...)

Should I open separate issues for these two problems?
